tap_youtube_analytics/streams/videos.py

```diff
@@ -0,0 +1,134 @@
+from typing import Any, Dict
+
+from singer import Transformer, get_logger, metrics, utils, write_bookmark, write_record
+from tap_youtube_analytics.streams.abstracts import IncrementalStream
+
+LOGGER = get_logger()
+
+
+class Videos(IncrementalStream):
+    tap_stream_id = "videos"
+    key_properties = ["id"]
+    replication_method = "INCREMENTAL"
+    replication_keys = ["published_at"]
+    data_key = "items"
+    path = "search"
+    endpoint = "search_videos"
+
+    def chunks(self, lst, cnt):
+        """Yield successive n-sized chunks from lst."""
+        for i in range(0, len(lst), cnt):
+            yield lst[i:i + cnt]
+
+    def sync(
+        self,
+        state: Dict,
+        transformer: Transformer,
+        parent_obj: Dict = None,
+    ) -> Dict:
+        """Incrementally sync videos for all configured channel IDs."""
+        self.url_endpoint = self.get_url_endpoint(parent_obj)
+        bookmark_date = self.get_bookmark(state, self.tap_stream_id)
+        last_dttm = utils.strptime_to_utc(bookmark_date)
+        current_max_bookmark_date = bookmark_date
+
+        channel_ids = self.client.config["channel_ids"]
+        channel_list = [cid.strip() for cid in channel_ids.split(",")]
+
+        total_written = 0
+
+        with metrics.record_counter(self.tap_stream_id) as counter:
+            for channel_id in channel_list:
+                search_params = {
+                    "part": "id,snippet",
+                    "channelId": channel_id,
+                    "order": "date",
+                    "type": "video",
+                    "maxResults": 50,
+                    "publishedAfter": bookmark_date,
+                }
+
+                self.path = "search"
+                self.endpoint = "search_videos"
+                self.data_key = "items"
+                self.params = search_params
+
+                search_records = self.get_records()
+                video_ids = set()
+
+                for search_record in search_records:
+                    video_id = search_record.get("id", {}).get("videoId")
+                    if not video_id:
+                        continue
+
+                    published_at = search_record.get("snippet", {}).get("publishedAt")
+                    if not published_at:
+                        continue
+
+                    try:
+                        search_record_dttm = utils.strptime_to_utc(published_at)
+                    except Exception:
+                        continue
+                    if search_record_dttm < last_dttm:
```

Comment:
Are we fetching records from YT in descending order? I don't see it mentioned in the code. Can you confirm if this order is guaranteed?